server/main.go

```diff
@@ -19,11 +19,52 @@ var dagList = make(map[string]*dag.Dag)
 
 func main() {
 	http.HandleFunc("/dag/add", addDagHandler)
+	http.HandleFunc("/dag/schedule", scheduleDagHandler)
 	fmt.Println("Server listening on :8080")
 	http.ListenAndServe(":8080", nil)
 }
 
-// Handler function for the "/api/hello" endpoint
+func scheduleDagHandler(w http.ResponseWriter, r *http.Request) {
+	clientId := r.Header.Get("Client-Id")
+	if clientId == "" {
+		http.Error(w, "Client-Id header is missing", http.StatusBadRequest)
+		return
+	}
+
+	dagName := r.Header.Get("dag-name")
+	if dagName == "" {
+		http.Error(w, "dag-name header is missing", http.StatusBadRequest)
+		return
+	}
+
+	if dag, ok := dagList[dagName]; ok {
+		vdag := &scheduler.VDag {
+			ClientId:      clientId,
+			DagDefinition: *dag,
+			PDagMap:       make(map[string][]*scheduler.PDagDeployment),
+		}
+
+		scheduler.Scheduler.ScheduleDag(vdag)
+
+		w.Header().Set("Content-Type", "application/json")
+		response := struct {
+			Message string `json:"message"`
+		}{
+			Message: "DAG scheduled successfully",
+		}
+
+		jsonResponse, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, "Error marshalling response", http.StatusInternalServerError)
+			return
+		}
+
+		w.Write(jsonResponse)
+	} else {
+		http.Error(w, "DAG not found", http.StatusNotFound)
```

Comment:
Add the name of the DAG trying to find

Reply:
if dag, ok := map; ok { 
   //some code
} else {
   // error code
   return
}

var dag
if dag, ok = map; !ok {
   //error code
   return 
}

//some code